gtsam/nonlinear/NonlinearFactor.cpp

```diff
@@ -106,7 +106,7 @@ double NoiseModelFactor::weight(const Values& c) const {
     if (noiseModel_) {
       const Vector b = unwhitenedError(c);
       check(noiseModel_, b.size());
-      return 0.5 * noiseModel_->weight(b);
```

Comment:
This might cause regressions. :slightly_frowning_face:

Reply:
So, this is very significant, and seems fishy. Too bad this stuff did not get resolved when it was fresh in our minds, but now we have to re-argue this, and why it was not in @yetongumich PR.

Reply:
I agree. I believe the `0.5` should NOT be present, since if the `noiseModel_` is not robust, the weight function should simply be 1 (which in NoiseModel.h it is). If the noise model is robust, then it should just return the weight value for that particular M-estimator given the residuals.

Reply:
Interestingly, there isn't any test for this :cry:

Reply:
Interestingly, there isn't any reasonable usages for this function either...

Also, I think the 0.5 factor should not be there. @dellaert

Reply:
The only usage of this function is in `gtsam_unstable/partition`, so I suggest this function to be moved to unstable as a standalone function there.

Plus, this change is correct.